pkg/connector/invitations.go

```diff
@@ -0,0 +1,246 @@
+	if err != nil {
+		l.Error("snyk-connector: delete-invitation: failed to delete invitation", zap.Error(err))
+		return outputAnnotations, err
+	}
+
+	return outputAnnotations, nil
+}
+
+func invitationResource(_ context.Context, invite *snyk.InviteResponseData, orgID string, parentID *v2.ResourceId) (*v2.Resource, error) {
+	return createInvitationResource(invite, orgID, parentID)
+}
+
+func createInvitationResource(invite *snyk.InviteResponseData, orgID string, parentID *v2.ResourceId) (*v2.Resource, error) {
+	profile := map[string]interface{}{
+		"email":  invite.Attributes.Email,
+		"role":   invite.Attributes.Role,
+		"org_id": orgID,
+		"status": "pending",
+	}
+
+	userTraitOptions := []rs.UserTraitOption{
+		rs.WithUserProfile(profile),
+		rs.WithEmail(invite.Attributes.Email, true),
+		rs.WithUserLogin(invite.Attributes.Email),
+		rs.WithStatus(v2.UserTrait_Status_STATUS_ENABLED),
+	}
+
+	resourceOptions := []rs.ResourceOption{}
+	if parentID != nil {
+		resourceOptions = append(resourceOptions, rs.WithParentResourceID(parentID))
+	}
+	resourceOptions = append(resourceOptions, rs.WithDescription(fmt.Sprintf("Invitation for %s with role %s in organization %s", invite.Attributes.Email, invite.Attributes.Role, orgID)))
+
+	resource, err := rs.NewUserResource(
+		fmt.Sprintf("Invitation: %s", invite.Attributes.Email),
+		invitationResourceType,
+		invite.ID,
+		userTraitOptions,
+		resourceOptions...,
+	)
+	if err != nil {
+		return nil, err
+	}
+
+	return resource, nil
+}
+
+// List returns all invitations from organizations as resource objects.
+func (i *invitationBuilder) List(ctx context.Context, parentResourceID *v2.ResourceId, _ *pagination.Token) ([]*v2.Resource, string, annotations.Annotations, error) {
+	if parentResourceID == nil {
+		return nil, "", nil, nil
+	}
+
+	// Only list invitations if parent is an organization
+	if parentResourceID.ResourceType != orgResourceType.Id {
+		return nil, "", nil, nil
+	}
+
+	orgID := parentResourceID.Resource
+
+	// Check if we should filter by orgs
+	if len(i.orgs) > 0 {
+		if _, ok := i.orgs[orgID]; !ok {
+			return nil, "", nil, nil
+		}
+	}
+
+	invites, err := i.client.ListInvites(ctx, orgID)
+	if err != nil {
+		return nil, "", nil, fmt.Errorf("snyk-connector: failed to list invitations: %w", err)
+	}
+
+	var rv []*v2.Resource
+	for _, invite := range invites {
+		// Skip inactive invitations (already accepted or revoked)
+		if !invite.Attributes.IsActive {
+			continue
+		}
+		inviteCopy := invite
+		resource, err := invitationResource(ctx, &inviteCopy, orgID, parentResourceID)
+		if err != nil {
+			return nil, "", nil, fmt.Errorf("snyk-connector: failed to create invitation resource: %w", err)
+		}
+
+		rv = append(rv, resource)
+	}
+
+	return rv, "", nil, nil
+}
+
+// Entitlements always returns an empty slice for invitations.
+func (i *invitationBuilder) Entitlements(_ context.Context, _ *v2.Resource, _ *pagination.Token) ([]*v2.Entitlement, string, annotations.Annotations, error) {
```

Comment:
nit: we can use the skip entitlements and grants annotation on this resource type